sympy/core/expr.py

```diff
@@ -160,7 +160,16 @@ def __pos__(self):
         return self
 
     def __neg__(self):
-        return Mul(S.NegativeOne, self)
+        # Mul has its own __neg__ routine so here, we just
+        # create a 2-args Mul with the -1 in the canonical
+        # slot 0
+        assert not self.is_Add
```

Comment:
Is this expected:
```
>>> x = symbols('x')
>>> -(x + 1)*(x - 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/core/expr.py", line 166, in __neg__
    assert not self.is_Add
AssertionError
```

Reply:
During the detection/correction phase of the work. There is a trap to detect negated Adds right now. You can get around this (for now) by writing `0-(x+1)`.